### ISSUE TITLE
Problems with ASR and the :native_or_unimrcp renderer

### ISSUE THREAD

**Issue by Jared-Prime** [adhearsion/punchblock#223]
Following up on a recent email and submitting an issue as suggested, I'd like to clarify what we need at Ifbyphone. 
## Preamble

We know we have these options available to us:
- recognizers `:unimrcp`, `:asterisk`
- renderers `:unimrcp`, `:native`, `:native_or_unimrcp`
- grammars `:voice`, `:dtmf`
## The Issue

Our defaults are a `:unimrcp` recognizer and a `:native_or_unimrcp` renderer. We use the `:native_or_unimrcp` renderer specifically to allow playback of audiofiles natively and TTS over unimrcp since [Neospeech doesn't support audio tags](https://github.com/cloudvox/apptastic/issues/144#issuecomment-25205538).

We don't get an input type that allows us to pass along the appropriate grammar.
## Our Approaches
1. We first tried sending a voice only grammar. The [switch defaults](https://github.com/adhearsion/punchblock/blob/develop/lib/punchblock/translator/asterisk/call.rb#L221-L233) to `ComposedPrompt`. Upon validating the options, we get an error saying voice is [not supported by default](https://github.com/adhearsion/punchblock/blob/develop/lib/punchblock/translator/input_component.rb#L52)
2. Then we tried changing the renderer to `:unimrcp`. This succeeds _but only without audio tags_.
3. Finally, we added to the [switch here](https://github.com/adhearsion/punchblock/blob/develop/lib/punchblock/translator/asterisk/call.rb#L232-L233) to allow `:native_or_unimrcp` to create an `MRCPPrompt`, and added to the [validator](https://github.com/adhearsion/punchblock/blob/develop/lib/punchblock/translator/asterisk/component/mrcp_prompt.rb#L14-L15) to have the renderer accepted. However, [it's not a great solution](https://github.com/adhearsion/punchblock/pull/222#issuecomment-43745955) and we'd still miss the fallback functionality.
## TL;DR

The easiest thing for us to do is to simply use a `:unimrcp` renderer; yet the rub of the issue is that the fix for TTS fallback does not work with ASR since [the fix only applies to `ComposedPrompt`](https://github.com/cloudvox/punchblock/commit/216d42b5f7b468e861e62e6d523b41a0030ad75c) and not also `MRCPPrompt`.

What do you suggest? We need both the fallback and ASR.
